spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -406,6 +407,7 @@ protected void afterUnbindProducer(ProducerDestination destination,
 			}
 			consumerEndpoint = createConsumerEndpoint(destination, group, properties);
 			consumerEndpoint.setOutputChannel(inputChannel);
+			applyBeanPostProcessors(consumerEndpoint, name);
 			this.consumerCustomizer.configure(consumerEndpoint, name, group);
 			if (consumerEndpoint instanceof InitializingBean) {
```

Comment:
If we use AutowireCapableBeanFactory#initializeBean can we also get rid of this call to `afterPropertiesSet()`, as it will get called at `AbstractAutowireCapableBeanFactory#invokeInitMethods`, followed by `applyBeanPostProcessorsAfterInitialization(...)`

Reply:
That's correct, but then we need to fix all the binders which call that `afterPropertiesSet()`

Reply:
> That's correct, but then we need to fix all the binders which call that afterPropertiesSet()


🤔 But `((InitializingBean) consumerEndpoint).afterPropertiesSet();` is called here already (and also in `doBindProducer`), and they are `final` methods.  

This change just removes it from these two methods and lets it get called via `applyInitializeBean` -> `initializeBean` -> `invokeInitMethods `instead.

Actually, `AbstractAutowireCapableBeanFactory#invokeInitMethods` does a load of security stuff I don't understand, but it looks like it is trying to achieve the same thing.

Reply:
Sorry, the above change is outdated now, but the same applies to the most recent commit

Reply:
That's correct. It is better to defer to Spring `ApplicationContext` to initialize bean properly rather than trying to do that manually. 
And that's why I said that we probably need to revise all the binders since we do that stuff correct already here on the core level.